docs/design/Cephfs-RDR-ConsistencyGroup.md

```diff
@@ -0,0 +1,411 @@
+(Regional Disaster Recovery). However, with the upcoming support
+for consistency groups in CephFS, Ramen can expand its capabilities
+to include consistency-group RDR for CephFS.
+
+## Motivation
+
+Consistency Groups are pivotal in maintaining data integrity during
+synchronizing related datasets across regions, mitigating
+the risk of data corruption or loss during failover events. They
+constitute essential components of Regional Disaster Recovery, enabling
+organizations to minimize risks and ensure the availability and integrity
+of their data across geographical locations.
+
+### Goals
+
+* Support Crash Consistency Group in RDR failover
+* Support Application Consistency Group in RDR relocate
+
+### Non-Goals
+
+* Consistency Group in MDR (Metro Disaster Recovery)
+* Consistency Group for RBD (Rados Block Device)
+
+## Design Details
+
+Currently, Ramen utilizes volsync to capture snapshots for
+each PVC in the application and synchronize the restored data
+to a remote destination. However, volsync lacks the capability
+to take a volume group snapshot for PVCs in a consistency group
+and sync the restored data to the destination.
+
+In this new design, Ramen will capture a volume group snapshot
+for the volumes within a consistency group and restore the
+snapshot to new read-only PVCs. Subsequently, Ramen will use
+volsync to synchronize the data in the read-only PVCs to the remote destination.
+
+### User Story
+
+For the application, users can choose to specify a
+consistency group for some or all of the PVCs:
+
+* The application can have one consistency group
+  that includes all PVCs.
+* The application can have one consistency group
+  where some PVCs belong to the group while others do not.
+* The application can have no consistency group,
+  meaning all PVCs are not part of any consistency group.
+
+### Does enabling the feature change any default behavior
+
+No, only PVCs included in a volume group will adhere to this design;
+other PVCs will continue to follow the current approach.
+
+### Dependencies
+
+* CephFS VolumeGroupSnapshot CSI
+
+### New API
+
+To support Volume Group during Regional DR,
+two additional CRDs have been introduced:
+
+* RGS (ReplicationGroupSource): This resource functions similarly
+  to volsync ReplicationSource, managing Volume Groups as a source.
+
+```
+type ReplicationGroupSource struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  Spec   ReplicationGroupSourceSpec   `json:"spec,omitempty"`
+  Status ReplicationGroupSourceStatus `json:"status,omitempty"`
+}
+
+type ReplicationGroupSourceSpec struct {
+  // Trigger to do the schedule sync or final sync
+  Trigger *ReplicationSourceTriggerSpec `json:"trigger,omitempty"`
+
+  VolumeGroupSnapshotClassName *string `json:"volumeGroupSnapshotClassName,omitempty"`
+
+  // RSSpecs specify the PVCs in a volume group
+  RSSpecs []VolSyncReplicationSourceSpec `json:"rsspecs,omitempty"`
```

Comment:
This should contain ReplicationSources

Reply:
Add the created ReplicationSources in the status.
- Spec.RSSpecs is the protectedPVCs show how to create the ReplicationSources
- Status.ReplicationSources show the created ReplicationSources reference